### PR TITLE
fix(security): add HTTPException guards and remove error.message leaks

### PR DESCRIPTION
**Pull request by amikofalvy**
## Summary

Fixes two bug classes across 4 route files (~10 catch blocks) identified in [PRD-6257](https://linear.app/inkeep/issue/PRD-6257):

- **Missing HTTPException guards** — Auth/validation `HTTPException`s thrown by middleware were caught by route-level catch blocks and re-thrown as generic 500s. Clients saw "internal server error" instead of the correct 401/403/400. Same bug class as Sentry #7322158580.
- **Raw `error.message` leaks** — Catch blocks passed `error.message` directly into `createApiError` for 500-level responses. DB errors like `connect ECONNREFUSED 10.0.0.5:5432` could reach the client verbatim.

### Files changed

| File | Fix |
|------|-----|
| `chatDataStream.ts` | Added HTTPException import + guard |
| `chat.ts` | Replaced weak shape check with `instanceof HTTPException`, removed `error.message` leak |
| `projectFull.ts` | Added/moved HTTPException guard in 4 catch blocks, replaced 4 message leaks |
| `agentFull.ts` | Added HTTPException import + guard in 3 catch blocks, replaced 3 message leaks |

### Tests added

- `chat.test.ts` — HTTPException pass-through (401), static message enforcement
- `agentFull.test.ts` — Safe 404 messages for GET/DELETE of non-existent agents
- `projectFull.test.ts` — Safe 404 messages for GET/DELETE of non-existent projects

## Test plan

- [x] `pnpm typecheck` passes
- [x] `pnpm lint:fix` clean
- [x] All 3 test suites pass (chat: 9, agentFull: 25, projectFull: 23)
- [ ] Verify auth failures (401/403) return correct status codes, not 500
- [ ] Verify DB connection errors produce static "Failed to..." messages, not raw error details

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: agents-api/src/__tests__/manage/integration/projectFull.test.ts [hunk 1]
```diff
@@ -600,6 +600,42 @@ describe('Project Full CRUD Routes - Integration Tests', () => {
...
+        `/manage/tenants/${tenantId}/project-full/${nonExistentId}`,
+        {
+          method: 'GET',
+          expectError: true,
+        }
+      );
+
+      expect(response.status).toBe(404);
+      const body = await response.json();
+      expect(body.detail).toContain('not found');
```
[the remark this answers is not in the document]
Comment (amikofalvy) [reply agents-api/src/__tests__/manage/integration/projectFull.test.ts]: implement the explicit check

File: agents-api/src/__tests__/manage/integration/projectFull.test.ts [hunk 2]
```diff
@@ -600,6 +600,42 @@ describe('Project Full CRUD Routes - Integration Tests', () => {
...
+      expect(body.detail).not.toContain('ECONNREFUSED');
```
[the remark this answers is not in the document]
Comment (amikofalvy) [reply agents-api/src/__tests__/manage/integration/projectFull.test.ts]: remove this here and the other line mentioned

File: agents-api/src/domains/run/routes/chatDataStream.ts [hunk 3]
```diff
@@ -548,6 +549,9 @@ app.openapi(chatDataStreamRoute, async (c) => {
       );
     });
   } catch (error) {
+    if (error instanceof HTTPException) {
+      throw error;
+    }
```
[the remark this answers is not in the document]
Comment (amikofalvy) [reply agents-api/src/domains/run/routes/chatDataStream.ts:554]: @claude add test coverage for this

**Comment by amikofalvy** [comment]
@claude address the Ito bot feedback. These edge cases appear to be valid

File: agents-api/src/__tests__/manage/integration/projectFull.test.ts [hunk 4]
```diff
@@ -600,6 +600,40 @@ describe('Project Full CRUD Routes - Integration Tests', () => {
...
+        `/manage/tenants/${tenantId}/project-full/${nonExistentId}`,
+        {
+          method: 'GET',
+          expectError: true,
+        }
+      );
+
+      expect(response.status).toBe(404);
+      const body = await response.json();
+      expect(body.detail).toBe('Project not found');
```
Comment (amikofalvy) [inline agents-api/src/__tests__/manage/integration/projectFull.test.ts]: @claude Look at the test failures related to this file. Clearly the explicit checks is not correct. The message is a pattern that contains the project id that was requested. Update the test assertion so that the tests pass

File: agents-api/src/domains/run/routes/chatDataStream.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (amikofalvy) [reply agents-api/src/domains/run/routes/chatDataStream.ts:554]: @pullfrog can you implement these changes and push the result
